src/t_cose_key.c

```diff
@@ -108,3 +110,56 @@ t_cose_key_decode(struct q_useful_buf_c cbor_encoded,
 
     return result;
 }
+
+/*
+ * Public function. See t_cose_key.h
+ */
+enum t_cose_err_t
+t_cose_key_encode(struct t_cose_key      key,
+                  struct q_useful_buf    key_buf,
+                  struct q_useful_buf_c *cbor_encoded)
+{
+    enum t_cose_err_t      result;
+    int32_t                cose_curve;
+    MakeUsefulBufOnStack(  x_coord_buf, T_COSE_BITS_TO_BYTES(T_COSE_ECC_MAX_CURVE_BITS));
+    MakeUsefulBufOnStack(  y_coord_buf, T_COSE_BITS_TO_BYTES(T_COSE_ECC_MAX_CURVE_BITS));
+    struct q_useful_buf_c  x_coord;
+    struct q_useful_buf_c  y_coord;
+    bool                   y_sign;
+    QCBOREncodeContext     cbor_encoder;
+    QCBORError             qcbor_result;
+
+    result = t_cose_crypto_export_ec2_key(key,
```

Comment:
I assume all the relevant checks from rfc9053 (point on curve, etc.) are done here, right?